doc/property.md

```diff
@@ -246,3 +246,104 @@ def test_coverage(capsys):
 # Assume
 Fault provides the function `f.assume` that uses the same interface as
 `f.assert_` and will generate a system verilog `assume` statement
+
+# Advanced Examples
+## Example 1
+### Verilog
+```verilog
+input [7:0] a, b, c;
+output [7:0] x, y;
+
+`ASSERT(name_A,
+  !$onehot(a) && |b && x[0] |=> y != $past(y, 2),
+ clk, resetn
+)
+```
+### Python
+```python
+class Foo(m.Circuit):
+    io = m.IO(a=m.In(m.Bits[8]), b=m.In(m.Bits[8]), c=m.In(m.Bits[8]),
+              x=m.Out(m.Bits[8]), y=m.Out(m.Bits[8]))
+    io += m.ClockIO(has_resetn=True)
+    # NOTE: Circuit code should appear above or else referring to output values
+    # (e.g. `io.y.value()` won't work)
+    # sva syntax
+    f.assert_(
+        f.sva(f.not_(f.onehot(io.a)), "&&",
```

Comment:
I've probably been out of the loop on is but: it seems like there are many different ways to do "logic" on signals. `&&` is represented as a string, `onehot` seems to be a verification (fault) specific function, and `|` (or-reduce) seems to be traditional magma logic. Curious why it's this way.

Reply:
"&&" is slightly different than "&" when it comes to precedence, but it's possible to do the equivalent logic with magma & rather than the "&&" sva syntax, I just chose to do it this way since it's based off an existing SVA example and it was closest to the original code (otherwise you have to make sure you get the parenthesis right to get the equivalent logic, which is what the infix example shows).  `onehot` is a fault function because there's no way to pass arguments to the sva operator strings, so we have to add Python level support for functions (the alternative would be doing something like `, "$onehot(", io.a, ")", ...` which seems undesirable, especially since we need the functions for the infix syntax anyways.

Reply:
(&& is similar to `and` in python (w.r.t. &), but as you know, we can't override that behavior)

Reply:
Got it. So it seems like constructing the asserts out of magma circuits is not really plausible.

Reply:
It would require downstream support.  E.g. we could have $onehot be a primitive that we instance (wire up in the inputs/outputs), but coreir would need to be aware of how to code generate the primitive (function call instead of structural).  coreir could add a similar `&&` operator, but here the issue is really a frontend/syntax problem (we could try a functional syntax like `f.and(a, b, c)`)